src/portfolios/portfolios.service.ts

```diff
@@ -78,4 +80,20 @@ export class PortfoliosService {
       transactionBaseDto
     );
   }
+
+  public async updatePortfolioName(
+    portfolioParams: PortfolioDto,
+    params: ModifyPortfolioDto
+  ): ServiceReturn<NumberedPortfolio> {
+    const { did, id } = portfolioParams;
+
+    if (id.lte(0)) {
+      throw new AppValidationError('Default portfolio name cannot be modified');
+    }
+
+    const { base, args } = extractTxBase(params);
+    const portfolio = (await this.findOne(did, id)) as NumberedPortfolio;
```

Comment:
Update `findOne` with this signature and you won't need to cast. Also mentioned [here](https://github.com/PolymeshAssociation/polymesh-rest-api/pull/163/files#r1081448512)

```
public async findOne(did: string): Promise<DefaultPortfolio>;
  public async findOne(did: string, portfolioId: BigNumber): Promise<NumberedPortfolio>;
  public async findOne(
    did: string,
    portfolioId?: BigNumber
  ): Promise<DefaultPortfolio | NumberedPortfolio> {
```

Reply:
Since its used in a few places I think the clean up is worth it. 

I don't like how type assertion looks with `await`, and TS provides a feature to solve this.